backend/src/main/java/com/huellapositiva/application/controller/ESALApiController.java

```diff
@@ -104,6 +105,51 @@ public void registerESAL(@Validated @RequestBody ESALRequestDto dto,
         }
     }
 
+
+
+    @Operation(
+            summary = "Return esal information",
+            description = "Return esal profile information",
+            tags = "ESAL",
+            parameters = {
+                    @Parameter(name = "X-XSRF-TOKEN", in = ParameterIn.QUERY, required = true, example = "a6f5086d-af6b-464f-988b-7a604e46062b", description = "For take this value, open your inspector code on your browser, and take the value of the cookie with the name 'XSRF-TOKEN'. Example: a6f5086d-af6b-464f-988b-7a604e46062b"),
+                    @Parameter(name = "XSRF-TOKEN", in = ParameterIn.COOKIE, required = true, example = "a6f5086d-af6b-464f-988b-7a604e46062b", description = "Same value of X-XSRF-TOKEN")
+            },
+            security = {
+                    @SecurityRequirement(name = "accessToken")
+            }
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "200",
+                            description = "Ok, return full esal information",
+                            content = {
+                                    @Content(
+                                            mediaType = "application/json",
+                                            schema = @Schema(implementation = GetESALResponseDto.class)
+                                    )
+                            }
+                    ),
+                    @ApiResponse(
+                            responseCode = "400",
+                            description = "Bad request, credentials are not valid",
+                            content = @Content(mediaType = "application/json")
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the esal data due to a connectivity issue.",
+                            content = @Content(mediaType = "application/json")
+                    )
+            }
+    )
+    @GetMapping
```

Comment:
GET /esal/{id} and check that the ContactPerson is the real owner of the ESAL

GET / is reserved to retrieve all resources or a paginated list

Consider that in the future a volunteer should also be able to retrieve public/limited set of information about an ESAL using the same endpoint.

fetchEsalAction.rexecuteAsOwner
fetchEsalAction.executeAsVolunteer

Reply:
We have created executeAsOwner for the Contact Person to be able to fetch and edit their ESAL information, but we haven't created other type of fetch yet, as we don't plan to have it included in the MVP.

Reply:
That makes total sense.